projects/concurrency/main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import "fmt"
+
+type Value[K comparable] struct {
+	data any
+	next *K
+	prev *K
+}
+
+type Cache[K comparable] struct {
+	size  int
+	m     map[K]Value[K]
+	first *K
+	last  *K
+}
+
+func NewCache[K comparable](entryLimit int) Cache[K] { //All K should be unique && same type && comparable type like primitive that can compare with each other
+	return Cache[K]{
+		size:  entryLimit,
+		m:     make(map[K]Value[K]),
+		first: nil,
+		last:  nil,
+	}
+}
+
+func (c *Cache[K]) Put(key K, value Value[K]) bool {
```

Comment:
Ideally the `Value` type wouldn't leak into your public interface. It's a useful implementation detail for you, but as far as a user is concerned they should ideally be able to just `c.Put("greeting", "hello")` without needing to know about your `Value` wrapping. Same with `Get` - it should return a pointer to a `string` or `any` or whatever, not a `Value`.